rust/lance-io/src/lib.rs

```diff
@@ -78,3 +78,86 @@ impl From<&Self> for ReadBatchParams {
         params.clone()
     }
 }
+
+impl ReadBatchParams {
+    /// Convert a read range into a vector of row offsets
+    ///
+    /// Can take in a `base_offset` and `length` to allow paging through the range
```

Comment:
I see. It slices and then materializes as a vector. 

Very much bikeshedding, but it almost feels more readable to do something like:

```rust
params.slice(base_offset, length).to_vec()
```

Also, doesn't it seem a little suboptimal that we need to now materialize these offsets when before we often just had ranges? Curious why we can't just keep as `ReadBatchParams`.

Reply:
I don't think we materialize more than before.  We have to materialize to assign row ids.  This logic existed already in the v1 reader.  Now it exists both in the v1 reader (for some legacy paths) and in the fragment (the new path).  Once the legacy paths are removed we can move this back into the fragment if we want.

Reply:
The slice method seems nice though.  I can do that.